### PR TITLE
Add high level filesystem documention

### PR DESCRIPTION
**Pull request by artonge**
Simple take over of https://github.com/nextcloud/server/pull/26982

> Adds some documentation for the filesystem layer, both a high level overview of how the various pieces interact and a high level guide for apps interacting with the filesystem.

> Hopefully this will be useful to anyone trying to either use the filesystem or work on the filesystem.

### CODE REVIEW

File: developer_manual/basics/storage/filesystem.rst [hunk 1]
```diff
@@ -1,37 +1,78 @@
 ==========
-Filesystem
+Nextcloud filesystem API
 ==========
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply developer_manual/basics/storage/filesystem.rst:3]:
```suggestion
========================
```

File: developer_manual/basics/storage/filesystem.rst [hunk 2]
```diff
@@ -1,37 +1,78 @@
...
```
Comment (nickvergessen) [inline developer_manual/basics/storage/filesystem.rst:1]:
```suggestion
========================
```

File: developer_manual/core/architecture/files.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
+============
+Nextcloud filesystem API
+============
```
Comment (nickvergessen) [inline developer_manual/core/architecture/files.rst:3]:
```suggestion
========================
Nextcloud filesystem API
========================
```

File: developer_manual/core/architecture/files.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            ▼                        ▼
+    ┌───────────────┐      ┌────────────────────┐
+    │Jail           ├─────►│CacheJail           │  Jail restricts access to a file or folder of a storage providing
+    └───────┬───────┘      └─────────┬──────────┘  a limited view into the storage (think unix chroot or bind mount)
+            │                        │
+            ▼                        ▼
+    ┌───────────────┐      ┌────────────────────┐
+    │Base Storage   ├─────►│Base Cache          │
+    └───────────────┘      └────────────────────┘
+Code Map
```
Comment (nickvergessen) [inline developer_manual/core/architecture/files.rst:89]: Explicit markup ends without a blank line; unexpected unindent.

```suggestion


Code Map
```

File: developer_manual/core/architecture/files.rst [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+--------
+
+Approximate overview of the significant filesystem code
+
+AppData
+~~~~~~~
```
Comment (nickvergessen) [inline developer_manual/core/architecture/files.rst:95]: Title level inconsistent:

and some more comments in github actions reply

File: developer_manual/core/architecture/files.rst [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+Code Map
+--------
+
+Approximate overview of the significant filesystem code
+
+AppData
+^^^^^^^
+
+High level api for accessing "appdata" folders, based on the ``Node`` API
```
Comment (CarlSchwan) [inline developer_manual/core/architecture/files.rst:100]:
```suggestion
High level api for accessing "appdata" folders, based on the ``Node/SimpleFs`` API
```

**Review by CarlSchwan — APPROVED** [review]
Nice stuff

**Pull request merged**